src/features/party/PartiesProvider.tsx

```diff
@@ -167,19 +182,35 @@ export function PartyProvider({ children }: PropsWithChildren) {
   );
 }
 
-export const usePartiesAllowedToInstantiate = () => usePartiesAllowedToInstantiateCtx();
+export const usePartiesAllowedToInstantiate = () => usePartiesAllowedToInstantiateCtx() ?? [];
+
+export const useCurrentParty = () => useCurrentPartyCtx().currentParty;
 
-/**
- * Returns the current party, or the custom selected current party if one is set.
- * Please note that the current party might not be allowed to instantiate, so you should
- * check the `canInstantiate` property as well.
- */
-export const useCurrentParty = () => useCurrentPartyCtx().party;
-export const useCurrentPartyIsValid = () => useCurrentPartyCtx().currentIsValid;
 export const useSetCurrentParty = () => useCurrentPartyCtx().setParty;
 
-export const useValidParties = () => usePartiesAllowedToInstantiateCtx()?.filter((party) => party.isDeleted === false);
+export const useValidParties = () => usePartiesAllowedToInstantiate().filter((party) => !party.isDeleted);
 
 export const useHasSelectedParty = () => useCurrentPartyCtx().userHasSelectedParty;
 
 export const useSetHasSelectedParty = () => useCurrentPartyCtx().setUserHasSelectedParty;
+
+export const useCurrentPartyIsValid = () => useCurrentPartyCtx().currentPartyIsValid;
+
+export function useInstanceOwnerParty() {
+  const instance = useLaxInstanceData((data) => data);
+  const parties = usePartiesAllowedToInstantiate();
```

Comment:
Some apps are always instantiated by the serviceowner themselves, in that case, do we risk the instance owner not showing up in this list? As in, a given instance owner may not have the right to instantiate
This is really common for some service owners so we need to be 100% sure.